tools/build.sh

```diff
@@ -130,22 +191,27 @@ monochrome_with_cache() {
   echo "${current_hash}" > "${cache_hash}"
 }
 
-# Handle drafts
-if [[ "${DRAFTS_MODE}" -eq 1 ]]; then
-  # shellcheck disable=SC2317
-  cleanup() {
-    if [[ -n "$original_dir" ]]; then
-      # Windows-specific cleanup
-      cd "$original_dir" || exit
-      git restore draft-scenarios/assets draft-scenarios/latexmkrc draft-scenarios/metadata.tex draft-scenarios/.version
-    fi
+# Handle scenario filtering, save for later use
+SCENARIO=""
+if [[ -n "${SCENARIO_KEYWORD}" ]]; then
+  if ! tools/find_scenario.sh "${SCENARIO_KEYWORD}"; then
+    exit 1
+  fi
+  SCENARIO=$(grep -o "[^/{}]*\.[^{}]*" structure.tex | cut -d'.' -f1)
```

Comment:
for mac support, this could be replaced with perl or awk or sed, I suppose

Reply:
I asked AI to make this cross-platform, and here are its proposals:
```sh
cat structure.tex | grep "[^/{}]*\.[^{}]*" | sed -E 's/.*[\/{}]([^\/{}]*)\.[^{}]*.*/\1/g'
perl -ne 'while (/[^\/{}]*\.[^{}]*/g) { print "$&\n"; }' structure.tex | awk -F. '{print $1}'
sed -n 's/.*[\/{}]\([^\/{}]*\)\.[^{}].*/\1/p' structure.tex
```
😵‍💫

Reply:
I chose Perl, as I suppose it's the only truly cross-platform tool in the mix. I have bad experience with both grep and sed on Mac.
```sh
perl -ne 'while(/[^\/{}]*\.[^{}]*/g){$_=$&;s/\..*//;print "$_\n"}' structure.tex
```